### ISSUE TITLE
Cannot drop a runtime in a context where blocking is not allowed while using Rocket with mailgun-rs

### ISSUE THREAD

**Issue by mjovanc** [dongri/mailgun-rs#6]
Hi, thanks for an awesome library first of. I tried setting it up, but I encounter some issues while using this with Rocket framework. I get this error:

```
thread 'rocket-worker-thread' panicked at 'Cannot drop a runtime in a context where blocking is not allowed. This happens when a runtime is dropped from within an asynchronous context.', /Users/marcus.cvjeticanin/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/runtime/blocking/shutdown.rs:51:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

This is my POST endpoint in Rocket that basically calls the function `send_mail_confirmation()` that is basically the `send_html()` function from your examples:

```rust
#[post("/order/save", format = "application/json", data = "<order>")]
pub fn save_order(order: Json<Order<'_>>) -> Json<MailConfirmation> {
    // take the order and use the information to send an email
    send_mail_confirmation(&order);

    let confirmation = MailConfirmation {
        message: "Mail sent successfully!",
    };
    Json(confirmation)
}
```

**Comment by dongri** [comment]
I add rocket examples to main branch
Use `tokio::task::block_in_place`
https://github.com/dongri/mailgun-rs/blob/main/examples/rocket/src/main.rs#L31-L43

# curl post
![Screenshot 2023-05-20 at 11 30 24](https://github.com/dongri/mailgun-rs/assets/878993/faa8cde8-1af0-428e-a317-6c33111b21f3)

# mail
![Screenshot 2023-05-20 at 11 30 46](https://github.com/dongri/mailgun-rs/assets/878993/8dc2bc6d-d6a8-4717-9e36-377b6764d60a)

**Comment by dongri** [comment]
I plan to add an async version of this library in the future

**Comment by dongri** [comment]
Released v0.1.6 
Added async_send

```rust
    match client.async_send(&sender).await {
        Ok(_) => {
            println!("successful");
        }
        Err(err) => {
            println!("Error: {err}");
        }
    }
```
https://github.com/dongri/mailgun-rs/blob/main/examples/rocket/src/main.rs#L106-L113

**Comment by mjovanc** [comment]
That's awesome. Thanks for so quick fix!
